### PR TITLE
feat(@desktop/chat): Save favorites and recents gif

### PR DESCRIPTION
**Pull request by alaibe**
Corresponding status go PR: https://github.com/status-im/status-go/pull/2296
image

### CODE REVIEW

File: ui/shared/status/StatusGifPopup.qml [hunk 1]
```diff
@@ -130,7 +218,7 @@ Popup {
             SVGImage {
                 id: gifImage
                 anchors.horizontalCenter: parent.horizontalCenter
-                source: "./assets/img/gif.svg"
+                source: Style.current.name === "light" ? "./assets/img/gif.svg": "./assets/img/gif-dark.svg"
```
Comment (richard-ramos) [inline ui/shared/status/StatusGifPopup.qml:221]:
```suggestion
                source: Style.current.name === Constants.lightThemeName ? "./assets/img/gif.svg": "./assets/img/gif-dark.svg"
```

Another option could be:
```
source: `./assets/img/gif-{Style.current.name}.svg`
```
and rename `gif.svg` to `gif-light.svg`

Comment (alaibe) [reply ui/shared/status/StatusGifPopup.qml]: Went for the second option, that is much nicer, thanks ;)

**Comment by richard-ramos** [comment]
This PR vendor/status-go is not pointing to the correct commit hash, causing this issue:

```
DBG 2021-08-09 18:43:16-04:00 callPrivateRPC                             topics="rpc" tid=2582780 file=core.nim:20 rpc_method=settings_saveSetting
Traceback (most recent call last)
/home/richard/status/status-desktop/src/nim_status_client.nim(240) nim_status_client
/home/richard/status/status-desktop/src/nim_status_client.nim(237) mainProc
/home/richard/status/status-desktop/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
/home/richard/status/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(49) qobjectCallback
/home/richard/status/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(34) onSlotCalled
/home/richard/status/status-desktop/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(574) onSlotCalled
/home/richard/status/status-desktop/src/app/chat/views/gif.nim(118) addToRecents
/home/richard/status/status-desktop/src/status/gif.nim(151) addToRecents
/home/richard/status/status-desktop/src/status/libstatus/gif.nim(15) setRecentGifs
/home/richard/status/status-desktop/src/status/libstatus/settings.nim(23) saveSetting
/home/richard/status/status-desktop/src/status/libstatus/core.nim(24) callPrivateRPC
ERR 2021-08-09 18:43:16-04:00 rpc response error                         topics="rpc" tid=2582780 file=core.nim:25 methodName=settings_saveSetting payload="[\"gifs/recent-gifs\",{\"items\":[{\"id\":\"4486363\",\"title\":\"????\",\"url\":\"https://media.tenor.com/images/86f3b65249fbaca12e142281558c06ac/tenor.gif\",\"tinyUrl\":\"https://media.tenor.com/images/7a059e812ebbac916587a247f96bda1a/tenor.gif\",\"height\":224}]}]" result="{\"jsonrpc\":\"2.0\",\"id\":0,\"error\":{\"code\":-32000,\"message\":\"configuration value not allowed\"}}"
Traceback (most recent call last)
```

**Review by richard-ramos — CHANGES_REQUESTED** [review]
This PR vendor/status-go is not pointing to the correct commit hash, causing this issue:
```
DBG 2021-08-09 18:43:16-04:00 callPrivateRPC                             topics="rpc" tid=2582780 file=core.nim:20 rpc_method=settings_saveSetting
Traceback (most recent call last)
/home/richard/status/status-desktop/src/nim_status_client.nim(240) nim_status_client
/home/richard/status/status-desktop/src/nim_status_client.nim(237) mainProc
/home/richard/status/status-desktop/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
/home/richard/status/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(49) qobjectCallback
/home/richard/status/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(34) onSlotCalled
/home/richard/status/status-desktop/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(574) onSlotCalled
/home/richard/status/status-desktop/src/app/chat/views/gif.nim(118) addToRecents
/home/richard/status/status-desktop/src/status/gif.nim(151) addToRecents
/home/richard/status/status-desktop/src/status/libstatus/gif.nim(15) setRecentGifs
/home/richard/status/status-desktop/src/status/libstatus/settings.nim(23) saveSetting
/home/richard/status/status-desktop/src/status/libstatus/core.nim(24) callPrivateRPC
ERR 2021-08-09 18:43:16-04:00 rpc response error                         topics="rpc" tid=2582780 file=core.nim:25 methodName=settings_saveSetting payload="[\"gifs/recent-gifs\",{\"items\":[{\"id\":\"4486363\",\"title\":\"????\",\"url\":\"https://media.tenor.com/images/86f3b65249fbaca12e142281558c06ac/tenor.gif\",\"tinyUrl\":\"https://media.tenor.com/images/7a059e812ebbac916587a247f96bda1a/tenor.gif\",\"height\":224}]}]" result="{\"jsonrpc\":\"2.0\",\"id\":0,\"error\":{\"code\":-32000,\"message\":\"configuration value not allowed\"}}"
Traceback (most recent call last)
```

**Comment by alaibe** [comment]
@richard-ramos I updated the branch with latest status-go.

**Review by richard-ramos — APPROVED** [review]
Approved and tested! good work!

**Pull request merged**